docs/reference/compute-engine/ray.md

```diff
@@ -0,0 +1,383 @@
+# Ray Compute Engine (contrib)
+
+The Ray compute engine is a distributed compute implementation that leverages [Ray](https://www.ray.io/) for executing feature pipelines including transformations, aggregations, joins, and materializations. It provides scalable and efficient distributed processing for both `materialize()` and `get_historical_features()` operations.
+
+## Overview
+
+The Ray compute engine provides:
+- **Distributed DAG Execution**: Executes feature computation DAGs across Ray clusters
+- **Intelligent Join Strategies**: Automatic selection between broadcast and distributed joins
+- **Lazy Evaluation**: Deferred execution for optimal performance
+- **Resource Management**: Automatic scaling and resource optimization
+- **Point-in-Time Joins**: Efficient temporal joins for historical feature retrieval
+
+## Architecture
+
+The Ray compute engine follows Feast's DAG-based architecture:
+
+```
+EntityDF → RayReadNode → RayJoinNode → RayFilterNode → RayAggregationNode → RayTransformationNode → Output
+```
+
+### Core Components
+
+| Component | Description |
+|-----------|-------------|
+| `RayComputeEngine` | Main engine implementing `ComputeEngine` interface |
+| `RayFeatureBuilder` | Constructs DAG from Feature View definitions |
+| `RayDAGNode` | Ray-specific DAG node implementations |
+| `RayDAGRetrievalJob` | Executes retrieval plans and returns results |
+| `RayMaterializationJob` | Handles materialization job tracking |
+
+## Configuration
+
+Configure the Ray compute engine in your `feature_store.yaml`:
+
+```yaml
+project: my_project
+registry: data/registry.db
+provider: local
+offline_store:
+    type: ray
+    storage_path: data/ray_storage
+batch_engine:
+    type: ray.engine
+    max_workers: 4                          # Optional: Maximum number of workers
+    enable_optimization: true               # Optional: Enable performance optimizations
+    broadcast_join_threshold_mb: 100       # Optional: Broadcast join threshold (MB)
+    max_parallelism_multiplier: 2          # Optional: Parallelism multiplier
+    target_partition_size_mb: 64           # Optional: Target partition size (MB)
+    window_size_for_joins: "1H"            # Optional: Time window for distributed joins
+    ray_address: localhost:10001           # Optional: Ray cluster address
+    use_ray_cluster: false                 # Optional: Use Ray cluster mode
+```
+
+### Configuration Options
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `type` | string | Required | Must be `ray.engine` |
```

Comment:
Is this table correct? You have "Required" as the value for the "Default" column. I think that means something may be wrong?